code/modules/sanity/sanity_mob.dm

```diff
@@ -0,0 +1,174 @@
+#define SANITY_PASSIVE_GAIN 0.2
+
+// Damage received from unpleasant stuff in view
+#define SANITY_DAMAGE_VIEW(damage, vig, dist) ((damage) * (1.2 - (vig) / STAT_LEVEL_MAX) * (1 - (dist)/15))
+
+#define SANITY_DAMAGE_THRESHOLD_VIEW 20
+
+// Damage received from body damage
+#define SANITY_DAMAGE_HURT(damage, vig) ((damage) / 5 * (1.2 - (vig) / STAT_LEVEL_MAX))
+
+// Damage received from shock
+#define SANITY_DAMAGE_SHOCK(shock, vig) ((shock) / 50 * (1.2 - (vig) / STAT_LEVEL_MAX))
+
+// Damage received from seeing someone die
+#define SANITY_DAMAGE_DEATH(vig) (10 * (1 - (vig) / STAT_LEVEL_MAX))
+
+#define SANITY_GAIN_SMOKE 0.05 // A full cig restores 300 times that
+#define SANITY_GAIN_SAY 1
+
+#define SANITY_COOLDOWN_SAY rand(30 SECONDS, 45 SECONDS)
+#define SANITY_COOLDOWN_BREAKDOWN rand(7 MINUTES, 10 MINUTES)
+
+
+/datum/sanity
+	var/flags
+	var/mob/living/carbon/human/owner
+
+	var/godmode = 0
+	var/level
+	var/max_level = 100
+
+	var/positive_prob = 20
+	var/negative_prob = 30
+
+	var/say_time = 0
+	var/breakdown_time = 0
+
+	var/list/datum/breakdown/breakdowns = list()
+
+/datum/sanity/New(mob/living/carbon/human/H)
+	owner = H
+	level = max_level
+	RegisterSignal(owner, COMSIG_HUMAN_SAY, .proc/onSay)
+
+/datum/sanity/proc/onLife()
+	var/affect = SANITY_PASSIVE_GAIN
+	if(owner.stat)
+		changeLevel(affect)
+		return
+	if(!(owner.sdisabilities & BLIND) && !owner.blinded)
+		affect += handle_area()
+		affect -= handle_view()
```

Comment:
Shouldn't it be _affect **+=** handle_view()_?

Reply:
No, `handle_view` returns damage.